app/routes/client.py

```diff
@@ -183,24 +192,28 @@ def event_archive(index):
 
     :param int index: The page to fetch
     """
-    index = int(index)
     if index <= 0:
         return redirect(url_for('.events'))
 
+    # Get all events that occur on this page or on subsequent pages
     today = date.today()
-    last_sunday = datetime.combine(today - timedelta(days=today.weekday()+7),
-                                   datetime.min.time())
+    events = (Event.objects(published=True, end_date__lt=today)
+                   .order_by('-start_date')
```

Comment:
I could either write inline explanations in a lot of places, or I could write a "Mongoengine in Eventum" markdown file documenting commonly used features.

What do you think is best?

(fwiw there is no other way to sort this way, without casting to `list` and reversing